core/src/main/scala/org/bitcoins/core/currency/CurrencyUnits.scala

```diff
@@ -6,15 +6,52 @@ import org.bitcoins.core.protocol.NetworkElement
 import org.bitcoins.core.serializers.RawSatoshisSerializer
 import org.bitcoins.core.util.Factory
 import scodec.bits.ByteVector
+import scala.math.Numeric
 
 sealed abstract class CurrencyUnit
     extends NetworkElement
     with Ordered[CurrencyUnit]
-    with BasicArithmetic[CurrencyUnit] {
+    with BasicArithmetic[CurrencyUnit]
+    with Numeric[CurrencyUnit] {
   type A
 
   def satoshis: Satoshis
 
+  override def plus(x: CurrencyUnit, y: CurrencyUnit): CurrencyUnit =
+    Satoshis(x.satoshis.underlying + y.satoshis.underlying)
+
+  override def minus(x: CurrencyUnit, y: CurrencyUnit): CurrencyUnit =
+    Satoshis(x.satoshis.underlying - y.satoshis.underlying)
+
+  override def times(x: CurrencyUnit, y: CurrencyUnit): CurrencyUnit =
+    Satoshis(x.satoshis.underlying * y.satoshis.underlying)
+
+  override def negate(x: CurrencyUnit): CurrencyUnit =
+    Satoshis(-x.satoshis.underlying)
+
+  override def fromInt(x: Int): CurrencyUnit = Satoshis(x.toLong)
+
+  override def parseString(str: String): Option[CurrencyUnit] = {
+    if (str.isEmpty) {
+      None
+    } else {
+      Some(Satoshis(str.toLong))
```

Comment:
`toLong` will throw if `str` is not correctly formatted long integer. Is it the behavior we want?

Reply:
This function was actually removed because it is no compatible with version `2.12.x`